src/sc/photogallery/browser/zip.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+from Products.Five.browser import BrowserView
+from Products.statusmessages.interfaces import IStatusMessage
+from sc.photogallery import _
+from zope.component import getMultiAdapter
+from ZPublisher.Iterators import filestream_iterator
+from sc.photogallery.utils import last_modified as utils_last_modified
+
+import os
+import pkg_resources
+
+try:
+    pkg_resources.get_distribution('ftw.zipexport')
+except pkg_resources.DistributionNotFound:
+    HAS_ZIPEXPORT = False
+else:
+    HAS_ZIPEXPORT = True
+
+if HAS_ZIPEXPORT:
+    from ftw.zipexport.generation import ZipGenerator
+    from ftw.zipexport.interfaces import IZipRepresentation
+
+
+class ZipView(BrowserView):
+    '''Export PhotoGallery as zip package'''
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+        self._path = []
+
+    @property
+    def traverse_subpath(self):
+        return self._path
+
+    def publishTraverse(self, request, name):
+        self._path.append(name)
+        return self
```

Comment:
@ericof could you please help me with this code? the publishTraverse method  is not working..